src/mechaphlowers/entities/arrays.py

```diff
@@ -0,0 +1,114 @@
+
+    def _compute_extra_columns(self, input_data: DataFrame) -> list[str]:
+        return [
+            column for column in input_data.columns if column not in self._input_columns
+        ]
+
+    def _drop_extra_columns(self, input_data: DataFrame) -> DataFrame:
+        """Return a copy of the input DataFrame, without irrelevant columns.
+
+        Note: This has no impact on the input DataFrame.
+        """
+        extra_columns = self._compute_extra_columns(input_data)
+        return input_data.drop(columns=extra_columns)
+
+    def __str__(self) -> str:
+        return self.data.to_string()
+
+
+class SectionInputDataFrame(pa.DataFrameModel):
+    """Schema for the data expected for a dataframe used to instantiate a SectionArray.
+
+    Each row describes a support and the following span (except the last row which "only" describes the last support).
+
+    Notes:
+        Line angles are expressed in degrees.
+
+        insulator_length should be zero for the first and last supports, since for now mechaphlowers
+        ignores them when computing the state of a span or section.
+        Taking them into account might be implemented later.
+
+        span_length should be numpy.nan for the last row.
+    """
+
+    name: Series[str]
+    suspension: Series[bool]
+    conductor_attachment_altitude: Series[float]
+    crossarm_length: Series[float]
+    line_angle: Series[float]
+    insulator_length: Series[float]
+    span_length: Series[float] = pa.Field(nullable=True)
+
+    @pa.dataframe_check(
+        description="""Though tension supports also have insulators,
+        for now we ignore them when computing the state of a span or section.
+        Taking them into account might be implemented later.
+        For now, set the insulator length to 0 for tension supports to suppress this error."""
+    )
+    def insulator_length_is_zero_if_not_suspension(cls, df: DataFrame) -> Series[bool]:
+        return (df["suspension"] | (df["insulator_length"] == 0)).pipe(Series[bool])
+
+    @pa.dataframe_check(
+        description="""Each row in the dataframe contains information about a support
+        and the span next to it, except the last support which doesn't have a "next" span.
+        So, specifying a span_length in the last row doesn't make any sense.
+        Please set span_length to "not a number" (numpy.nan) to suppress this error.""",
+    )
+    def no_span_length_for_last_row(cls, df: DataFrame) -> bool:
+        return df.tail(1)["span_length"].isna().all()
+
+
+class SectionArray(ElementArray):
+    """Description of an overhead line section.
+
+    Args:
+        data: Input data
+        sagging_parameter: Sagging parameter
+        sagging_temperature: Sagging temperature, in Celsius degrees
+    """
+
+    @pa.check_types(lazy=True)
+    def __init__(
+        self,
+        data: DataFrame[SectionInputDataFrame],
+        sagging_parameter: float,
+        sagging_temperature: float,
```

Comment:
@adriengoeller I changed the code so that sagging parameter and temperature can be provided after initialization, while keeping the possibility to set it at initialization. 
For example, both following code snippets would be valid:

```python
section = SectionArray(input_df)
...
section.sagging_parameter = 2_000
section.sagging_temperature = 15
```
or

```python
section = SectionArray(
    input_df,
    sagging_parameter=2_000,
    sagging_temperature=15,
)
```

I didn't use a `@property` decorator for this after all, but we may still use one in the future if needed.

If it's OK for you, I let you resolve this conversation.